connection.go

```diff
@@ -28,82 +21,48 @@ type DatabendConn struct {
 	cfg         *Config
 	cancel      context.CancelFunc
 	closed      int32
-	stmts       []*databendStmt
 	logger      *log.Logger
 	rest        *APIClient
 	batchMode   bool
 	batchInsert func() error
 }
 
 func (dc *DatabendConn) exec(ctx context.Context, query string, args ...driver.Value) (driver.Result, error) {
-	respCh := make(chan QueryResponse)
-	errCh := make(chan error)
-	dc.rest.NextQuery()
-
-	go func() {
-		err := dc.rest.QuerySync(ctx, query, args, respCh)
-		errCh <- err
-	}()
-
-	for {
-		select {
-		case err := <-errCh:
-			if err != nil {
-				return emptyResult, err
-			} else {
-				return emptyResult, nil
-			}
-		case resp := <-respCh:
-			b, err := json.Marshal(resp.Data)
-			if err != nil {
-				return emptyResult, err
-			}
-			_, _ = io.Copy(io.Discard, bytes.NewReader(b))
-		}
+	_, err := dc.rest.QuerySync(ctx, query, args)
```

Comment:
We need to use `for-select` to wait the `QuerySync` until it end.

Reply:
now waiting inside QuerySync, as it`s name implies.